src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1138 @@
+     * @throws Exception during http communication, except 403
+     */
+    private void retrieveSystemStatus(Map<String, String> statistics) throws Exception {
+        ArrayNode response = doGet(STATUS, ArrayNode.class);
+        response.iterator().forEachRemaining(jsonNode -> {
+            String langtag = getJsonProperty(jsonNode, "langtag", String.class);
+            ArrayNode stateList = getJsonProperty(jsonNode, "stateList", ArrayNode.class);
+
+            if (langtag != null && stateList != null) {
+                statistics.put("System Status#" + normalizeStringPropertyName(
+                        langtag.replaceAll("_", " ")),
+                        stateList.get(0).asText().replaceAll("_", " ").toUpperCase());
+            }
+        });
+    }
+
+    /**
+     * Change property name to start with a capital letter
+     *
+     * @param string parameter name
+     * @return {@link String} property name with first capital letter
+     */
+    private String normalizeStringPropertyName(String string) {
+        return string.substring(0, 1).toUpperCase() + string.substring(1).toLowerCase();
+    }
+
+    /**
+     * Retrieve the current state of the collaboration session
+     *
+     * @return Integer value of the current device volume level
+     * @throws Exception during http communication
+     */
+    private void retrieveCollaborationStatus(Map<String, String> statistics) throws Exception {
+        JsonNode response = doGet(COLLABORATION, JsonNode.class);
+        if (!response.isNull()) {
+            String sessionState = getJsonProperty(response, "state", String.class);
+            statistics.put("Collaboration#Session State", sessionState);
+            if ("ACTIVE".equals(sessionState)) {
+                statistics.put("Collaboration#Session ID", getJsonProperty(response, "id", String.class));
+            }
+        }
+    }
+
+    /**
+     * Retrieve current volume level of the device
+     *
+     * @return Integer value of the current device volume level
+     * @throws Exception during http communication
+     */
+    private Integer retrieveVolumeLevel() throws Exception {
+        return doGet(VOLUME, Integer.class);
+    }
+
+    /**
+     * Update device's volume with a given value
+     * Since we need to react to the volume level change to have {@link #localStatistics} data synchronized with
+     * the actual values that are sent here - we need to set new status for the locally stored control properties.
+     * This is why response code is being checked for the operation.
+     * Also this specific request doesn't have any response body, so the only piece of data that we can use to verify
+     * the request is response code.
+     *
+     * @param value target value for the device volume
+     * @throws Exception during http communication
+     */
+    private void updateVolumeLevel(int value) throws Exception {
+        if(doPost(VOLUME, value).getStatusCode().is2xxSuccessful()) {

```

Comment:
This change is weird. Control operation failed and there is no error? Has to throw exception if failed.
Also I still did not get clear answer on that is2xxSuccessful check. Isn't one of the base classes already checking HTTP response codes and throws exceptions if they are not success codes? If yes - this check here is redundant. If not - then why only in this method?